src/Utils.ts

```diff
@@ -986,12 +986,129 @@ function setDummyInputDebugValue(
+    private _domChangedTimer?: number;
+    private _dummyMap: WeakMap<HTMLElement, () => void> = new WeakMap();
+    private _dummies: WeakRef<HTMLElement>[] = [];
+    domChanged?(parent: HTMLElement | null): void;
+
+    constructor(win: GetWindow) {
+        this._win = win;
+    }
+
+    add(dummy: HTMLElement | undefined, callback: () => void): void {
+        if (dummy) {
+            const dummyMap = this._dummyMap;
+            const cb = dummyMap.get(dummy);
+
+            if (!cb) {
+                this._dummies.push(new WeakRef(dummy));
+                this.domChanged = this._domChanged;
+            }
+
+            dummyMap.set(dummy, callback);
+        }
+    }
+
+    remove(dummy: HTMLElement | undefined): void {
+        if (dummy) {
+            const dummyMap = this._dummyMap;
+            const cb = dummyMap.get(dummy);
+
+            if (cb) {
+                dummyMap.delete(dummy);
+
+                const dummies = this._dummies;
+
+                for (let i = dummies.length; i--; ) {
+                    const el = dummies[i].deref();
+
+                    if (el) {
+                        if (el === dummy) {
+                            dummies.splice(i, 1);
+                            break;
+                        }
+                    } else {
+                        dummies.splice(i, 1);
+                    }
+                }
+
+                if (dummies.length === 0) {
+                    delete this.domChanged;
+                }
+            }
+        }
+    }
+
+    dispose(): void {
+        const win = this._win?.();
+
+        if (this._offsetsTimer) {
+            win?.clearTimeout(this._offsetsTimer);
+            delete this._offsetsTimer;
+        }
+
+        if (this._domChangedTimer) {
+            win?.clearTimeout(this._domChangedTimer);
+            delete this._domChangedTimer;
+        }
+
+        this._domChangedMap = new WeakMap();
+        this._dummyMap = new WeakMap();
+        this._dummies = [];
+
+        delete this._win;
+    }
+
+    private _domChanged = (parent: HTMLElement | null): void => {
+        if (parent && !this._domChangedMap.has(parent)) {
+            this._domChangedMap.set(parent, true);
+
+            if (!this._domChangedTimer) {
+                this._domChangedTimer = this._win?.().setTimeout(() => {
+                    delete this._domChangedTimer;
+
+                    for (const dummyRef of this._dummies) {
+                        const dummy = dummyRef.deref();
+                        const dummyParent = dummy?.parentElement;
+
+                        if (
+                            dummyParent &&
+                            this._domChangedMap.has(dummyParent)
+                        ) {
+                            this._dummyMap.get(dummy)?.();
+                        }
+                    }
+
+                    this._domChangedMap = new WeakMap();
+                }, 100);
+            }
+        }
```

Comment:
```suggestion
        if (!parent || this._domChangedMap.has(parent)) {
          return
        }
        
        this._domChangedMap.set(parent, true);

        if (this._domChangedTimer) {
          return
        }

        this._domChangedTimer = this._win?.().setTimeout(() => {
            delete this._domChangedTimer;

            for (const dummyRef of this._dummies) {
                const dummy = dummyRef.deref();
                const dummyParent = dummy?.parentElement;

                if (
                    dummyParent &&
                    this._domChangedMap.has(dummyParent)
                ) {
                    this._dummyMap.get(dummy)?.();
                }
            }

            this._domChangedMap = new WeakMap();
        }, 100);
```

avoid adding another level of nesting by short returning (the less nested the easier to read)